src/pdm/cli/commands/build.py

```diff
@@ -57,29 +58,54 @@ def do_build(
-                artifacts.append(sdist_file)
-            if wheel:
+            if not project.config["use_uv"]:
                 if sdist:
-                    project.core.ui.echo("[info]Building wheel from sdist...")
-                    sdist_out = tempfile.mkdtemp(prefix="pdm-build-via-sdist-")
-                    try:
-                        with tarfile.open(sdist_file, "r:gz") as tf:
-                            tf.extractall(sdist_out)
-                            sdist_name = os.path.basename(sdist_file)[: -len(".tar.gz")]
-                            whl = WheelBuilder(os.path.join(sdist_out, sdist_name), project.environment).build(dest)
-                            project.core.ui.echo(f"[info]Built wheel at {whl}")
-                            artifacts.append(whl)
-                    finally:
-                        shutil.rmtree(sdist_out, ignore_errors=True)
-                else:
-                    project.core.ui.echo("[info]Building wheel...")
-                    whl = WheelBuilder(project.root, project.environment).build(dest)
-                    project.core.ui.echo(f"[info]Built wheel at {whl}")
-                    artifacts.append(whl)
+                    project.core.ui.echo("[info]Building sdist...")
+                    sdist_file = SdistBuilder(project.root, project.environment).build(dest)
+                    project.core.ui.echo(f"[info]Built sdist at {sdist_file}")
+                    artifacts.append(sdist_file)
+                if wheel:
+                    if sdist:
+                        project.core.ui.echo("[info]Building wheel from sdist...")
+                        sdist_out = tempfile.mkdtemp(prefix="pdm-build-via-sdist-")
+                        try:
+                            with tarfile.open(sdist_file, "r:gz") as tf:
+                                tf.extractall(sdist_out)
+                                sdist_name = os.path.basename(sdist_file)[: -len(".tar.gz")]
+                                whl = WheelBuilder(os.path.join(sdist_out, sdist_name), project.environment).build(dest)
+                                project.core.ui.echo(f"[info]Built wheel at {whl}")
+                                artifacts.append(whl)
+                        finally:
+                            shutil.rmtree(sdist_out, ignore_errors=True)
+                    else:
+                        project.core.ui.echo("[info]Building wheel...")
+                        whl = WheelBuilder(project.root, project.environment).build(dest)
+                        project.core.ui.echo(f"[info]Built wheel at {whl}")
+                        artifacts.append(whl)
+            else:
+                import subprocess
+
+                dist_dir = project.root / "dist"
+                dest_dir = Path(dest).absolute()
+
+                subprocess.call(["uv", "build"])
+
+                (dist_dir / ".gitignore").unlink(missing_ok=True)
+                if dest_dir != dist_dir:
```

Comment:
Can be removed

Reply:
> Can be removed

`(dist_dir / ".gitignore").unlink(missing_ok=True)` is reserved for the consistency with `use_uv false`, and `pdm publish` would fail with .gitignore